tests/integrations/langchain/langchain_nv_ai_endpoints_test.py

```diff
@@ -0,0 +1,543 @@
+import os
+
+import pytest
+from langchain_core.messages import AIMessageChunk
+from langchain_nvidia_ai_endpoints import ChatNVIDIA
+
+import weave
+from weave.integrations.integration_utilities import op_name_from_ref
+
+model = "meta/llama-3.1-8b-instruct"
+
+
+@pytest.mark.skip_clickhouse_client  # TODO:VCR recording does not seem to allow us to make requests to the clickhouse db in non-recording mode
+@pytest.mark.vcr(
+    filter_headers=["authorization"],
+    allowed_hosts=["api.wandb.ai", "localhost"],
+)
+def test_chatnvidia_quickstart(client: weave.trace.weave_client.WeaveClient) -> None:
+    api_key = os.environ.get("NVIDIA_API_KEY", "DUMMY_API_KEY")
+
+    nvidia_client = ChatNVIDIA(
+        api_key=api_key, model=model, temperature=0.0, max_tokens=64, top_p=1
+    )
+
+    response = nvidia_client.invoke("Hello!")
+
+    calls = list(client.calls())
+    # need to make 2 because of the langchain integration getting a call in there
+    assert len(calls) == 2
+    call = calls[1]
+
+    assert response.content is not None
+
+    assert op_name_from_ref(call.op_name) == "langchain.Llm.ChatNVIDIA-generate"
```

Comment:
surprised this name passes

Reply:
weave name sanitization turns it into this